core/src/main/scala/org/apache/spark/internal/config/UI.scala

```diff
@@ -0,0 +1,120 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.internal.config
+
+import java.util.concurrent.TimeUnit
+
+import org.apache.spark.network.util.ByteUnit
+
+private[spark] object UI {
+
+  val UI_SHOW_CONSOLE_PROGRESS = ConfigBuilder("spark.ui.showConsoleProgress")
```

Comment:
I'm leaning a bit on the side of removing the `UI_` prefix here, and always referencing these as `UI.foo` in other code.

But don't really feel strongly either way.

Reply:
Sounds nice. I'd like to ensure "how to use configurations" be consistent though: breaking down config package object into multiple (like what we do with UI, History, Kafka, etc) and referring configuration as `<object name>.<configuration name>` all the cases. 

Would it make sense to file issues for cleanup (more groups, change the way to refer configurations) as part of [SPARK-26442](https://issues.apache.org/jira/browse/SPARK-26442)?

Reply:
Someone filed SPARK-22194 a while ago that is in that same spirit.

As for consistency, it may be better to be even more explicit (e.g. `config.Blah.CONSTANT`) but that is pretty verbose. Some parts of the code already do that.